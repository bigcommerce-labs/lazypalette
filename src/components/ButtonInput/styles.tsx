import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
