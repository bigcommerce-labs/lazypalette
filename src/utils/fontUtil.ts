export function parseFont(font: string) {
    const provider = font.split('_')[0];
    switch (provider) {
        case 'Google':
            return parseGoogleFont(font);
        default:
            return null;
    }
}

function parseGoogleFont(font: string) {
    const split = font.split('_');
    const family = split[1];
    let formattedFont = '';
    let weight = split[2];

    if (split.length === 2) {
        formattedFont += `${family}|`;
    } else if (split.length > 2) {
        weight = weight.split(',')[0];
        formattedFont += `${family}:${weight}|`;
    }

    return `//fonts.googleapis.com/css?family=${formattedFont}`;
}
