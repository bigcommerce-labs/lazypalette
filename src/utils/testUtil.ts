export const sel = (id: string) => `[data-test-id="${id}"]`;
