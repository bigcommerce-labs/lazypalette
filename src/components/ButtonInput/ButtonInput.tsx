import React, { MouseEvent, MouseEventHandler, PureComponent } from 'react';

import { Button, Wrapper } from './styles';

interface ButtonInputProps extends Partial<{
    border: boolean;
    children: JSX.Element | string;
    classType: string;
    disabled: boolean;
    onClick: MouseEventHandler<HTMLButtonElement>;
    testId?: string;
    type: string;
}> {}

class ButtonInput extends PureComponent<ButtonInputProps> {
    handleClick = (e: MouseEvent<HTMLButtonElement>) => {
        if (this.props.onClick) {
            this.props.onClick(e);
        }
    };

    render() {
        const {
            border = true,
            disabled,
            classType,
            testId,
            type,
            children,
        } = this.props;

        return (
            <Wrapper>
                <Button
                    border={border}
                    disabled={disabled}
                    onClick={this.handleClick}
                    classType={classType}
                    type={type}
                    {...(testId ? {'data-test-id': testId} : {})}
                >
                    {children}
                </Button>
            </Wrapper>
        );
    }
}

export default ButtonInput;
