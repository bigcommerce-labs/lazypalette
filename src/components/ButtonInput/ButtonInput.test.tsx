import { shallow } from 'enzyme';
import React from 'react';

import { sel } from '../../utils/testUtil';

import ButtonInput from './ButtonInput';

describe('<ButtonInput />', () => {

    const testId = 'buttonInput';

    it('renders', () => {
        const buttonInput = shallow(
            <ButtonInput disabled={false} classType="primary" type="button">
          Click Meow!
            </ButtonInput>
        );

        expect(buttonInput).toMatchSnapshot();
    });

    it('onClick handler works', () => {
        const mockClick = jest.fn();
        const event = {
            target: {
                value: 'testing',
            },
        };
        const buttonInput = shallow(
            <ButtonInput
                disabled={false}
                classType="primary"
                type="button"
                onClick={mockClick}
                testId={testId}
            >
          Click Meow!
            </ButtonInput>
        );

        buttonInput.find(sel(testId)).simulate('click', event);
        expect(mockClick).toBeCalledWith(event);
    });

});
