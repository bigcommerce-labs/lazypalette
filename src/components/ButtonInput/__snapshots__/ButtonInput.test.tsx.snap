// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonInput /> renders 1`] = `
<styled.div>
  <styled.button
    border={true}
    classType="primary"
    disabled={false}
    onClick={[Function]}
    theme={
      Object {
        "colors": Object {
          "background": "#F5F7FA",
          "brandPrimary": "#273A8A",
          "primary": "#3F69FE",
        },
        "typography": Object {
          "fontFamily": Object {
            "sans": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Arial, sans-serif",
          },
        },
      }
    }
    type="button"
  >
    Click Meow!
  </styled.button>
</styled.div>
`;
